src/base/utils/foreignapps.cpp

```diff
@@ -142,50 +141,55 @@ namespace
 
         const DWORD cBuffer = (cbData / sizeof(WCHAR)) + 1;
-        LPWSTR lpData = new WCHAR[cBuffer]{0};
+        LPWSTR lpData = new WCHAR[cBuffer] {0};
         [[maybe_unused]] const auto lpDataGuard = qScopeGuard([&lpData] { delete[] lpData; });
 
         const LSTATUS res = ::RegQueryValueExW(handle, nameWStr.c_str(), NULL, &type, reinterpret_cast<LPBYTE>(lpData), &cbData);
         if (res == ERROR_SUCCESS)
-            return QString::fromWCharArray(lpData);
+            return Path(QString::fromWCharArray(lpData));
 
         return {};
     }
 
-    QStringList pythonSearchReg(const REG_SEARCH_TYPE type)
+    PathList pythonSearchReg(const REG_SEARCH_TYPE type)
     {
         const HKEY hkRoot = (type == USER) ? HKEY_CURRENT_USER : HKEY_LOCAL_MACHINE;
         const REGSAM samDesired = KEY_READ
             | ((type == SYSTEM_64BIT) ? KEY_WOW64_64KEY : KEY_WOW64_32KEY);
-        QStringList ret;
+        PathList ret;
 
         HKEY hkPythonCore {0};
         if (::RegOpenKeyExW(hkRoot, L"SOFTWARE\\Python\\PythonCore", 0, samDesired, &hkPythonCore) == ERROR_SUCCESS)
         {
             [[maybe_unused]] const auto hkPythonCoreGuard = qScopeGuard([&hkPythonCore] { ::RegCloseKey(hkPythonCore); });
 
             // start with the largest version
-            QStringList versions = getRegSubkeys(hkPythonCore);
+            PathList versions = getRegSubkeys(hkPythonCore);
             // ordinary sort won't suffice, it needs to sort ["3.9", "3.10"] correctly
-            std::sort(versions.begin(), versions.end(), Utils::Compare::NaturalLessThan<Qt::CaseInsensitive>());
+            const Utils::Compare::NaturalCompare<Qt::CaseInsensitive> comparator;
+            std::sort(versions.begin(), versions.end(), [&comparator](const Path &left, const Path &right)
+            {
+                return comparator(left.data(), right.data());
+            });
+
             ret.reserve(versions.size() * 2);
 
             while (!versions.empty())
             {
-                const std::wstring version = QString(versions.takeLast() + u"\\InstallPath").toStdWString();
+                const std::wstring version = (versions.takeLast() / Path(u"InstallPath"_s)).toString().toStdWString();
 
                 HKEY hkInstallPath {0};
                 if (::RegOpenKeyExW(hkPythonCore, version.c_str(), 0, samDesired, &hkInstallPath) == ERROR_SUCCESS)
                 {
                     [[maybe_unused]] const auto hkInstallPathGuard = qScopeGuard([&hkInstallPath] { ::RegCloseKey(hkInstallPath); });
 
-                    const QString basePath = getRegValue(hkInstallPath);
+                    const Path basePath = getRegValue(hkInstallPath);
                     if (basePath.isEmpty())
                         continue;
 
-                    if (const QString path = (basePath + u"python3.exe"); QFile::exists(path))
+                    if (const Path path = (basePath / Path(u"python3.exe"_s)); path.exists())
```

Comment:
@Chocobo1 
Don't you think "python.exe" and "python3.exe" are good candidates to be declared as named constants? They are repeated too many times in this code.

Reply:
>Don't you think "python.exe" and "python3.exe" are good candidates to be declared as named constants? They are repeated too many times in this code.

I don't feel strongly. Two times isn't too much for me.

Reply:
> > Don't you think "python.exe" and "python3.exe" are good candidates to be declared as named constants? They are repeated too many times in this code.
> 
> I don't feel strongly. Two times isn't too much for me.

It seemed much more to me... Perhaps I was just confused by reviewing code changes. Nevermind.